addons/hr_holidays/models/hr_employee.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+# Copyright (c) 2005-2006 Axelor SARL. (http://www.axelor.com)
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class HrEmployee(models.Model):
+    _inherit = "hr.employee"
+
+    remaining_leaves = fields.Float(compute='_compute_remaining_days', string='Remaining Legal Leaves', inverse='_inverse_remaining_days', help='Total number of legal leaves allocated to this employee, change this value to create allocation/leave request. Total based on all the leave types without overriding limit.')
+    current_leave_state = fields.Selection([('draft', 'New'), ('confirm', 'Waiting Approval'),
+        ('refuse', 'Refused'), ('validate1', 'Waiting Second Approval'),
+        ('validate', 'Approved'), ('cancel', 'Cancelled')],
+        compute='_compute_leave_status', string="Current Leave Status")
+    current_leave_id = fields.Many2one('hr.holidays.status', compute='_compute_leave_status', string="Current Leave Type")
+    leave_date_from = fields.Date(compute='_compute_leave_status', string='From Date')
+    leave_date_to = fields.Date(compute='_compute_leave_status', string='To Date')
+    leaves_count = fields.Integer(compute='_compute_leaves_count', string='Number of Leaves')
+    show_leaves = fields.Boolean(compute='_compute_show_leaves', string="Able to see Remaining Leaves")
+    is_absent_totay = fields.Boolean(compute='_compute_absent_employee', search='_search_absent_employee', string="Absent Today")
+
+    def _remaining_leaves_count(self):
+        leaves = self.env['hr.holidays'].read_group([('state', '=', 'validate'),
+            ('holiday_status_id.limit', '=', False), ('employee_id', 'in', self.ids)],
+            ['number_of_days', 'employee_id'], ['employee_id'])
+        leaves = dict([(leave['employee_id'][0], leave['number_of_days']) for leave in leaves])
+        return dict([(employee.id, leaves.get(employee.id, 0)) for employee in self])
```

Comment:
There's no need to create a list to pass to a dict, you can just pass in a generator comprehension to the dict, or even better just use a dictionary comprehension.